# Determination - Deterministic rendering environment for music and art
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, version 3.

# For some reason ZynAddSubFX's slot numbers change when compiled with the GUI
# disabled. That causes audio exported with ZynAddSubFX compiled without the
# GUI to be different than if compiled with any GUI. This is a really hacky
# fix that makes the slot numbers when compiled without a GUI the same as when
# compiled with a GUI by making DISTRHO Plugin Framework think we compiled the
# plugin with a GUI. It should be fine since we never use the ZynAddSubFX GUI in
# this image.

--- a/DPF/distrho/src/DistrhoPluginLV2.cpp
+++ b/DPF/distrho/src/DistrhoPluginLV2.cpp
@@ -685,7 +685,7 @@ public:
 
         updateParameterOutputsAndTriggers();
 
-#if DISTRHO_PLUGIN_WANT_STATE && DISTRHO_PLUGIN_HAS_UI
+#if false
         fEventsOutData.initIfNeeded(fURIDs.atomSequence);
 
         LV2_Atom_Event* aev;
--- a/src/Plugin/ZynAddSubFX/DistrhoPluginInfo.h
+++ b/src/Plugin/ZynAddSubFX/DistrhoPluginInfo.h
@@ -22,7 +22,7 @@
  #define DISTRHO_PLUGIN_HAS_UI          1
  #define DISTRHO_PLUGIN_HAS_EMBED_UI    1
  #define DISTRHO_PLUGIN_HAS_EXTERNAL_UI 1
-#elif defined(FLTK_GUI)
+#elif defined(DISABLE_GUI) || defined(FLTK_GUI)
  #define DISTRHO_PLUGIN_HAS_UI          1
  #define DISTRHO_PLUGIN_HAS_EMBED_UI    0
  #define DISTRHO_PLUGIN_HAS_EXTERNAL_UI 1
